Sources/LLBuildManifest/DOTManifestSerializer.swift

```diff
@@ -0,0 +1,65 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import TSCBasic
+
+/// Serializes an LLBuildManifest graph to a .dot file
+public struct DOTManifestSerializer {
+    var kindCounter = [String: Int]()
+    var hasEmittedStyling = Set<String>()
+    let manifest: BuildManifest
+
+    /// Creates a serializer that will serialize the given manifest.
+    public init(manifest: BuildManifest) {
+        self.manifest = manifest
+    }
+
+    /// Gets a unique label for a job name
+    mutating func label(for command: Command) -> String {
+        let toolName = "\(type(of: command.tool).name)"
+        var label = toolName
+        if let count = kindCounter[label] {
+            label += " \(count)"
+        }
+        kindCounter[toolName, default: 0] += 1
+        return label
+    }
+
+    /// Quote the name and escape the quotes
+    func quoteName(_ name: String) -> String {
+        return "\"" + name.replacingOccurrences(of: "\"", with: "\\\"") + "\""
```

Comment:
If the name might contain backslashes (which is admittedly unlikely), then they would need to be escaped too.